src/netman/dbus.go

```diff
@@ -137,3 +141,32 @@ func Ssids() ([]SSID, map[string]string, map[string]string) {
 	SSIDs := getSSIDs(conn, APs, ssid2ap)
 	return SSIDs, ap2device, ssid2ap
 }
+
+func ConnectedWifi() bool {
+	conn := getSystemBus()
+	objPath := dbus.ObjectPath("/org/freedesktop/NetworkManager")
+	nm := conn.Object("org.freedesktop.NetworkManager", objPath)
+	var nmConnectivityStatus uint32
+	err := nm.Call("org.freedesktop.NetworkManager.CheckConnectivity", 0).Store(&nmConnectivityStatus)
+	if err != nil {
+		fmt.Printf("error connected(): %v\n", err)
+		return false
+	}
+	if nmConnectivityStatus == 1 {
+		fmt.Println("Connected.")
+		return false //not connected by any netman interfaces
+	}
+	return true
+}
```

Comment:
good catch. this is happening because none of your interfaces are managed by network manager at the time of execution. code should do the right thing. I'll fix this.